cocos/2d/CCSpriteFrameCache.cpp

```diff
@@ -352,36 +356,33 @@ void SpriteFrameCache::addSpriteFramesWithDictionary(ValueMap& dict, const std::
 
 void SpriteFrameCache::addSpriteFramesWithFile(const std::string& plist, Texture2D *texture)
 {
-    if (_loadedFileNames->find(plist) != _loadedFileNames->end())
+    if (_spriteFramesCache.hasPlist(plist))
     {
         return; // We already added it
     }
     
     std::string fullPath = FileUtils::getInstance()->fullPathForFilename(plist);
     ValueMap dict = FileUtils::getInstance()->getValueMapFromFile(fullPath);
 
-    addSpriteFramesWithDictionary(dict, texture);
-    _loadedFileNames->insert(plist);
+    addSpriteFramesWithDictionary(dict, texture, plist);
 }
 
 void SpriteFrameCache::addSpriteFramesWithFileContent(const std::string& plist_content, Texture2D *texture)
 {
     ValueMap dict = FileUtils::getInstance()->getValueMapFromData(plist_content.c_str(), static_cast<int>(plist_content.size()));
-    addSpriteFramesWithDictionary(dict, texture);
+    addSpriteFramesWithDictionary(dict, texture, "by#addSpriteFramesWithFileContent()");
```

Comment:
`by#addSpriteFramesWithFileContent()` is used as plist?

Reply:
yes, it is not a valid filename, and only used as a `key` of index which will not be accessed.